src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -460,6 +480,23 @@ public boolean exists() {
         }
     }
 
+    /**
+     * Determines if the file is read-only.
+     *
+     * @return <tt>true</tt> if the file is read-only, <tt>false</tt> otherwise
+     */
+    public boolean readOnly() {
+        try {
+            if (readOnlyFlagSupplier != null) {
```

Comment:
**Optional:** Two new methods in the file, one uses the if as a guard-case, one the other way around. maybe switch one of them to be more consistent?
Other methods (as far as I can see in the PR view) seem to follow the way of the method I commented on.